### PR TITLE
checksum-compare: Add version 1.42

### PR DESCRIPTION
**Pull request by kunago**

### CODE REVIEW

File: bucket/checksum-compare.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            "Checksum compare.exe",
+            "Checksum compare"
+        ]
+    ],
+    "checkver": {
+        "url": "http://checksumcompare.sanktuaire.com/downloads",
+        "regex": "\\(Version Actuelle : v([\\d.]+)\\)"
+    },
+    "autoupdate": {
+        "url": "http://checksumcompare.sanktuaire.com/download.php?file=portable&version=v$version"
```
Comment (Ash258) [inline bucket/checksum-compare.json:25]:
```suggestion
        "url": "http://sanktuaire.free.fr/checksumcompare/Checksum-compare-v$version.7z",
        "hash": {
            "url": "http://checksumcompare.sanktuaire.com/downloads-en",
            "regex": "(?sm)$basename.*?$sha1"
        }
```

File: bucket/checksum-compare.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    ],
+    "shortcuts": [
+        [
+            "Checksum compare.exe",
+            "Checksum compare"
+        ]
+    ],
+    "checkver": {
+        "url": "http://checksumcompare.sanktuaire.com/downloads",
+        "regex": "\\(Version Actuelle : v([\\d.]+)\\)"
```
Comment (Ash258) [inline bucket/checksum-compare.json:22]:
```suggestion
        "url": "http://checksumcompare.sanktuaire.com/downloads-en",
        "regex": "Checksum-compare-v([\\d.]+)\\.7z"
```

File: bucket/checksum-compare.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            "checksum-compare"
+        ]
+    ],
+    "shortcuts": [
+        [
+            "Checksum compare.exe",
+            "Checksum compare"
+        ]
+    ],
+    "checkver": {
```
Comment (Ash258) [inline bucket/checksum-compare.json:20]:
```suggestion
    "persist": "Checksum compare.ini",
    "checkver": {
```

File: bucket/checksum-compare.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+    "version": "1.42",
+    "description": "Calculates and checks SHA1/MD5 checksums.",
+    "license": "Freeware",
+    "homepage": "http://checksumcompare.sanktuaire.com",
+    "hash": "e13ff69013a554bb8c33e428783e860f5ae0567db7db13aab323cb0043900547",
+    "url": "http://checksumcompare.sanktuaire.com/download.php?file=portable&version=v1.42#/dl.7z",
```
Comment (Ash258) [inline bucket/checksum-compare.json:7]:
```suggestion
    "homepage": "http://checksumcompare.sanktuaire.com/home",
    "license": "Freeware",
    "url": "http://sanktuaire.free.fr/checksumcompare/Checksum-compare-v1.42.7z",
    "hash": "sha1:4f32ff3ed69e78de98e9060b7293c09ee2c8ceee",
    "pre_install": [
        "if (!(Test-Path \"$persist_dir\\Checksum compare.ini\")) {",
        "    Set-Content \"$dir\\Checksum compare.ini\" '[MainWindow]', 'lang=en' -Encoding Ascii",
        "}"
    ],
```

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/checksum-compare.json (hunk 1 above)
Comment (kunago) [reply bucket/checksum-compare.json:25]: Why this change when the original URL is from the official website?

Comment (Ash258) [reply bucket/checksum-compare.json:31]: This is from oficial page too... Why would you have URL with redirect, when you can have static, direct URL??

Comment (kunago) [reply bucket/checksum-compare.json:31]: image

It does not redirect me or maybe I did not notice.

Comment (Ash258) [reply bucket/checksum-compare.json]: ... 


image

Comment (kunago) [reply bucket/checksum-compare.json]: Okay, no problem. Cannot commit the suggestion now though as it is outdated.

**Pull request merged**